src/gurobi_optimods/metromap.py

```diff
@@ -0,0 +1,1240 @@
+"""
+Metro Map: Computing an Octilinear Graph Representation
+-------------------------------------------------------
+"""
+
+import logging
+import math
+
+import gurobipy as gp
+import pandas as pd
+
+try:
+    import networkx as nx
+except ImportError:
+    nx = None
+
+try:
+    import plotly.graph_objs as go
+    from plotly.subplots import make_subplots
+except ImportError:
+    go = None
+
+try:
+    import plotly.io as pio
+
+    pio.renderers.default = "browser"
+except ImportError:
+    pio = None
+
+
+from gurobi_optimods.utils import optimod
+
+logger = logging.getLogger(__name__)
+
+
+@optimod()
+def metromap(
+    graph,
+    linepath_data=None,
+    include_planarity=True,
+    penalty_edge_directions=1,
+    penalty_line_bends=1,
+    penalty_distance=1,
+    *,
+    create_env,
+):
+    """Compute a metromap drawing, i.e., an octilinear network representation
+     considering the origin geographical data if available given as geographic position of the nodes
+
+    Parameters
+    ----------
+    graph : networkx graph with node attribute 'pos' being a tuple of x- and y-coordinate
```

Comment:
This should follow numpydoc's style: https://numpydoc.readthedocs.io/en/latest/format.html#parameters

It expects the argument name to be followed by a type, and the description should follow on the next line. `graph : Graph` would keep the checker happy and link to the networkx docs for the Graph class.